sdk/src/user.ts

```diff
@@ -1046,21 +1053,31 @@ export class User {
 	 */
 	public getMaxLeverage(
 		marketIndex: number,
+		marketType: MarketType,
 		category: MarginCategory = 'Initial'
 	): BN {
-		const market = this.driftClient.getPerpMarketAccount(marketIndex);
+		const marketIsPerp = isVariant(marketType, 'perp');
+
+		const market = marketIsPerp
+			? this.driftClient.getPerpMarketAccount(marketIndex)
+			: this.driftClient.getSpotMarketAccount(marketIndex);
 
 		const totalAssetValue = this.getTotalAssetValue();
 		if (totalAssetValue.eq(ZERO)) {
 			return ZERO;
 		}
 
+		// if spot trade w/ no margin, max lev = 1
+		if (!marketIsPerp && !this.getUserAccount().isMarginTradingEnabled) {
+			return ONE;
+		}
+
 		const totalLiabilityValue = this.getTotalLiabilityValue();
 
 		const marginRatio = calculateMarketMarginRatio(
```

Comment:
for spot, i think we'd want to get the liability weight with `calculateLiabilityWeight` and then you'd dive the free collateral by the liability weight.

it might make sense for getMaxLeverage to fork into `getMaxLeverageForPerp` and `getMaxLeverageForSpot`